src/dbn_struct.jl

```diff
@@ -0,0 +1,222 @@
+"""
+    DBNVar(x::Symbol)
+    DBNVar{x::Symbol}()
+
+Reference to a named node in the POMDP or MDP dynamic Bayesian network (DBN).
+
+Note that `gen(::DBNVar, m, depargs..., rng)` always takes an argument for each dependency, `gen(::DBNOut, m, s, a, rng)` only takes `s` and `a` arguments (the inputs to the entire DBN).
+
+`DBNVar` is a "value type". See the documentation of `Val` for more conceptual details about value types.
```

Comment:
Link to the doc here? https://docs.julialang.org/en/v1/manual/types/index.html#%22Value-types%22-1

Reply:
done